### PR TITLE
Branch refactor code

### LINKED ISSUE #88: separating command class
Current command class encapsulates everything.

Propose to change to one abstract command class.

Then create specific command class that inherits from abstract command and just run .execute
image

### PR DESCRIPTION
**Pull request by shxr3f**
This was a long one. Basically implemented Command classes for existing commands. I have commented out the Appointment part @tanweili can accomodate the new features based on the current structure. A lot has been done. I have yet to move all the printing to UI yet, will require someones help. Please look through the code see if it makes sense! Resolves #88

### CODE REVIEW

**Comment by Anvitha-r** [comment]
looks good i think the code seems neater and more organised

File: src/main/java/seedu/duke/helper/Command.java [hunk 1]
```diff
@@ -2,147 +2,153 @@
...
-        try {
-            String[] findAppointmentParameters = Parser.parseFindAppointment(parameters);
-            String criteria = findAppointmentParameters[0];
-            String input = findAppointmentParameters[1];
-            appointmentList.find(criteria, input);
-        } catch (HalpmiException e) {
-            UI.printParagraph(e.toString());
-        }
-    }
+//    private boolean isNull(String string) {
```
Comment (Anvitha-r) [inline src/main/java/seedu/duke/helper/Command.java]: can delete this commented code?

**Review by tanweili — APPROVED** [review]
Okay and understand the new general structure of the code. I'll add in the appt stuff in the new structure then. And also deleting by nric

**Review by Anvitha-r — APPROVED** [review]

File: src/main/java/seedu/duke/helper/Parser.java [hunk 2]
```diff
@@ -23,22 +33,80 @@
...
+    }
+
+    public static Command parseEditDoctor(String parameters) throws HalpmiException {
+        return null;
+    }
+
+
+    public static Command parseAddMedicine(String parameters) throws HalpmiException {
+        String[] medicineParameters = minParameterCheck(parameters, 6);
+        Validator.validateMedicine(medicineParameters);
```
Comment (tanweili) [inline src/main/java/seedu/duke/helper/Parser.java:89]: Can make an if statment with the return boolean value of the validator

**Review by tanweili — COMMENTED** [review]
Some parse methods in Parser may need to include an if statement to make the boolean return value of the respective validate methods in each Parser method useful.

**Pull request merged**